messagequeues/DataCaptureService.cs

```diff
@@ -0,0 +1,87 @@
+﻿
+using System;
+using System.IO;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Tesseract;
+using System.Drawing;
+
+namespace messagequeues;
+
+public class DataCapture
+{
+   public DataCapture()
+   {
+        ConnectionFactory factory = new();
+        factory.Uri = new Uri("amqp://guest:guest@localhost:5672");
+        factory.ClientProvidedName = "DataCaptureService Sender App";
+
+        IConnection cnn = factory.CreateConnection();
+        IModel channel = cnn.CreateModel();
+
+        string exchangeName = "DataCaptureService";
+        string routingKey = "demo-key";
+        string queueName = "DataCaptureServiceQueue";
+
+        channel.ExchangeDeclare(exchangeName, ExchangeType.Direct);
+        channel.QueueDeclare(queueName, false, false, false, null);
+        channel.QueueBind(queueName, exchangeName, routingKey, null);
+
+        byte[] imageBytes = File.ReadAllBytes("test.jpg");
+        Console.WriteLine("Image Uploaded"); 
+        channel.BasicPublish(exchangeName, routingKey, null, imageBytes);

```

Comment:
What if the file size is more than the message size limit in rabbitmq? One of the home task challenges is to deal with big files

![image](https://github.com/Sinancay/Learn/assets/43816231/6b4906ec-c0ea-4a73-820c-25dda7cfba9c)